elf_reader.go

```diff
@@ -750,7 +750,12 @@ func (ec *elfCode) loadBTFMaps() error {
 			// Each Var representing a BTF map definition contains a Struct.
 			mapStruct, ok := v.Type.(*btf.Struct)
```

Comment:
I think using https://pkg.go.dev/github.com/cilium/ebpf/btf#UnderlyingType here is more appropriate.